[WIP]fix #5079 fini g+

artragis opened the pull request:
```
 QA

- Aller sur la page de login
- on vous propose d'utiliser votre compte google et non google+

Nota: il me manque simplement le jeu d'icônes
```

**EDIT Helmasaur**

Remplace l'icône et le nom de Google+ à Google.

Numéro du ticket concerné : close #5079

### Contrôle qualité

1. Être déconnecté ;
2. Aller sur la page d'authentification ;
3. Vérifier que le bouton en dessous de celui de Facebook affiche le dernier logo de Google te le libellé « Google » au lieu de « Google+ » ;
4. Aller sur la d'inscription ;
5. Faire la même vérification que 3.

A-312 commented:
Il faut télécharger la version SVG, la recolorer et la save en png au format voulu.

artragis commented:
elle est où la version svg?

A-312 commented:
https://fr.m.wikipedia.org/wiki/Fichier:Google_%22G%22_Logo.svg ?

helmasaur committed:
🍱 Ajout des icônes de Google

helmasaur committed:
💄 Modification du bouton Google+ en bouton Google

helmasaur commented:
J'ai repris la PR.

Les images ont été ajoutées et les modifications ont été appliqués au libellé et au logo des deux boutons. Par contre, je me demande s'il faudrait changer la couleur du fond du bouton. La couleur la plus souvent utilisée est le blanc pour ce bouton. La couleur actuelle est celle de Google+, mais il me semble qu'il s'agit de la couleur rouge du logo Google. On peut donc la garder ou change pour [une des trois autres couleurs de Google](https://usbrandcolors.com/google-colors).

Aussi, faut-il supprimer ce qui est en rapport à Google+ (images et SCSS) ?

artragis commented:
utilise du blanc, ça devrait être mieux.

helmasaur commented:
Le problème avec le blanc est le contraste non ? Le bouton que je vois habituellement a une ombre.

![bouton-google](https://user-images.githubusercontent.com/873762/88091118-e66e3c00-cb8e-11ea-84bc-d783e7a6507c.png)

artragis commented:
Ok, donc faut probablement changer l'écriture

helmasaur commented:
Moi, je veux bien mais @AmauryCarrade va râler par rapport à l'accessibilité 😄.

AmauryCarrade commented:
Google impose [des règles assez strictes](https://developers.google.com/identity/branding-guidelines) sur le style du bouton « Se connecter avec Google ». Actuellement, on ne les respecte pas.

- L'icône n'est pas la bonne (il _faut_ le **G** multicolore sur fond blanc ; la version monochrome est explicitement bannie).
- Le texte non plus.
- Je n'ai pas vérifié les marges mais il faudrait.
- Google n'autorise que son blanc ou son bleu (cf. page liée pour les codes hexa des couleurs), donc le rouge est à proscrire. 

Enfin, si on reste sur le blanc, il faudra changer la couleur du texte (sinon je vais _effectivement_ m'opposer à la fusion en l'état).

helmasaur commented:
J'avais vu pour les règles de Google mais si on veut les respecter, pas sûr que l'on respecte ceux de Twitter ni de Facebook non plus.

- L'icône de la capture d'écran est de la prod. J'ai mis la nouvelle en trois versions (gris, gris clair et bleu).
- J'ai laissé le texte tel quel pour que ça soit uniforme
- /
- J'avais pensé au bleu mais ça se rapproche du bouton de Facebook et le blanc ne ressort pas vraiment du fond

Pour être franc, je ne suis pas contre respecter les règles des sites mais franchement, ça ne passe avec tous les designs. Ce n'est pas possible d'avoir un design avec ce genre de bouton qui sort totalement de la charte graphique du site pour faire plaisir à Google ou je ne sais quel autre site. Par contre, je pensais que c'était le fond du bouton et le fond du site qui ne contrastait suffisamment non plus.

EDIT :

Même l'exemple de Facebook ne tient pas la route dans al réalité.

- Idéal : https://developers.facebook.com/docs/facebook-login/userexperience/?locale=fr_FR#showvaluefirst
- Réalité (être déconnecté de Pinterest et cliquer sur le bouton  « S'inscrire » en haut à droite) : https://www.pinterest.fr

AmauryCarrade commented:
Google [se réserve le droit de couper l'accès à la connexion via eux](https://www.google.com/permissions/trademark/brand-terms/) si on ne respecte pas ces règles et qu'ils s'en rendent compte (ils avertissent avant, cela dit, d'expérience). C'est aussi nécessaire si on veut avoir une connexion via eux “certifiée” (si ce n'est pas déjà le cas), mais ça encore, c'est moins important il me semble.

> L'icône de la capture d'écran est de la prod. J'ai mis la nouvelle en trois versions (gris, gris clair et bleu).

L'icône que je vois dans le diff n'est pas la bonne (elle est explicitement interdite par les règles de Google).

> J'ai laissé le texte tel quel pour que ça soit uniforme.

On peut uniformiser avec la tournure de Google s'ils y tiennent (cela dit je ne pense pas que ce soit le plus important de leurs règles).

> J'avais pensé au bleu mais ça se rapproche du bouton de Facebook et le blanc ne ressort pas vraiment du fond.
> […] Par contre, je pensais que c'était le fond du bouton et le fond du site qui ne contrastait suffisamment non plus.

On peut mettre une bordure autour du bouton éventuellement ? Mais je ne sais pas trop ce que ça rendrait.

---

Il faudrait _au moins_ utiliser le bon logo et la bonne couleur de fond.

helmasaur commented:
Au pire, je mets la bouton en « bleu Google » pour au moins supprimer la présence de Google+ et je travaille sur une prochaine PR pour le respect de la charte graphique de Facebook et de Google ?

AmauryCarrade commented:
J'ai pris dix minutes pour récupérer les bons logos et respecter a peu près les règles [pour Google](https://developers.google.com/identity/branding-guidelines) (tout est respecté sauf les polices) et [pour Facebook](https://developers.facebook.com/docs/facebook-login/userexperience/#buttondesign) (tout est respecté, ils sont beaucoup moins casse pieds à tout imposer) — histoire qu'on avance ici.

@Helmasaur, tu as régulièrement demandé de l'aide sur cette PR cet été mais sans vraiment préciser sur quoi, donc je suis parti du principe que je pouvais tenter quelque chose. Ça donne ça (anciens boutons dessous pour comparaison).

![Boutons de connexion avec Google et Facebook](https://user-images.githubusercontent.com/1417570/92327256-a4268e80-f058-11ea-8592-c49a9f44a016.png)

AmauryCarrade committed:
Boutons de connexion avec les bons designs

- Ajout des icônes de Google et Facebook actualisées au format SVG.
- Suppression des sprites de Google+, Google, Foursquare (??).
- Modification des boutons de connexion avec les couleurs et les règles de
  Google et Facebook.

helmasaur commented:
Je ne demandais pas de l'aide, mais de valider la PR en attendant que je retravaille dessus pour que ça respecte les conventions de Google et de Facebook. Comme ça n'a pas été fait. Je n'ai pas avancé plus.

J'aurais préféré qu'on valide cette PR qui date d'il y a plusieurs semaines, et qui aurait pu être inclus au précédent patch, plutôt que de faire ce que je comptais faire…

Je trouve ça gonfler de dire « histoire qu'on avance » alors que je n'ai cessé de rappeler l'existence de cette PR.

AmauryCarrade commented:
Ah pardon, on s'est mal compris ! Je ne commit rien alors (de toute façon ça m'a pris si peu de temps) pour te laisser le faire. Par contre je pense vraiment, vu le temps que ça m'a demandé, que ça gagnerait à être fait directement.

Tu ne faisais que dire « c'est possible de s'occuper de ma PR concernant Google+ ? », ce qui peut s'interpréter de bien des façons… Tu n'as _jamais_ dit que tu attendais validation, et en terme de validation j'avais déjà répondu : le logo et la couleur n'étaient pas les bonnes. Donc je ne pensais pas que c'était l'objet de ta demande. J'avais aussi plussoyé le message précédent, en terme de validation, même si vu d'aujourd'hui je pense qu'il serait mieux de tout faire directement.

Je te laisse juste histoire que tu gagnes du temps sans avoir à fouiller les pages de règles de Google et Facebook : 
- les couleurs de Google : `hsl(217, 89%, 61%)` (survol : `hsl(217, 89%, 31%)`, fond blanc permanent obligatoire `hsl(0, 0%, 100%)`) ;
- les couleurs de Facebook : `hsl(214, 89%, 52%)` (survol : `hsl(214, 89%, 32%)`) ;
- les icônes qu'il faut utiliser (versions à jour ; version blanche pour Facebook, comme au dessus).

<details><summary>Icônes Google & Facebook</summary>

Facebook

```svg
<svg version="1.1" viewBox="0 0 1365.3 1365.3" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"><metadata><rdf:RDF><cc:Work rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"/><dc:title/></cc:Work></rdf:RDF></metadata><g transform="matrix(1.3333 0 0 -1.3333 0 1365.3)"><g transform="scale(.1)"><path d="m10240 5120c0 2827.7-2292.3 5120-5120 5120s-5120-2292.3-5120-5120c0-2555.5 1872.3-4673.7 4320-5057.8v3577.8h-1300v1480h1300v1128c0 1283.2 764.38 1992 1933.9 1992 560.17 0 1146.1-100 1146.1-100v-1260h-645.62c-636.03 0-834.38-394.67-834.38-799.57v-960.43h1420l-227-1480h-1193v-3577.8c2447.7 384.1 4320 2502.3 4320 5057.8" fill="#fff"/></g></g></svg>
```

Google

```svg
<svg viewBox="0 0 533.5 544.3" xmlns="http://www.w3.org/2000/svg"><path d="M533.5 278.4c0-18.5-1.5-37.1-4.7-55.3H272.1v104.8h147c-6.1 33.8-25.7 63.7-54.4 82.7v68h87.7c51.5-47.4 81.1-117.4 81.1-200.2z" fill="#4285f4"/><path d="M272.1 544.3c73.4 0 135.3-24.1 180.4-65.7l-87.7-68c-24.4 16.6-55.9 26-92.6 26-71 0-131.2-47.9-152.8-112.3H28.9v70.1c46.2 91.9 140.3 149.9 243.2 149.9z" fill="#34a853"/><path d="M119.3 324.3c-11.4-33.8-11.4-70.4 0-104.2V150H28.9c-38.6 76.9-38.6 167.5 0 244.4l90.4-70.1z" fill="#fbbc04"/><path d="M272.1 107.7c38.8-.6 76.3 14 104.4 40.8l77.7-77.7C405 24.6 339.7-.8 272.1 0 169.2 0 75.1 58 28.9 150l90.4 70.1c21.5-64.5 81.8-112.4 152.8-112.4z" fill="#ea4335"/></svg>
```

</details>

---

> Je trouve ça gonfler de dire « histoire qu'on avance » alors que je n'ai cessé de rappeler l'existence de cette PR.

Je disais ça pour moi (et les autres), en fait. Vu que _justement_ personne n'avait pris le temps de traiter ça. Par contre je tiens à rappeler que je fais ça bénévolement sur mon temps libre. J'ai tout de même le droit d'être en vacances l'été, et de choisir quand je travaille sur quoi…

Situphen commented:
> Je ne demandais pas de l'aide, mais de valider la PR en attendant que je retravaille dessus pour que ça respecte les conventions de Google et de Facebook.

Je pense qu'il est préférable que l'on attende que cette PR respecte les conventions de Google et de Facebook avant de la fusionner. Il n'y a rien d'urgent à enlever la mention de Google+ à mon avis donc je propose d'attendre que tu aies le temps de continuer à travailler dessus et on fusionnera plus tard ! :)

Situphen commented:
> Je ne demandais pas de l'aide, mais de valider la PR en attendant que je retravaille dessus pour que ça respecte les conventions de Google et de Facebook. Comme ça n'a pas été fait. Je n'ai pas avancé plus.

Il y a deux petites choses qu'il faudrait faire avant de valider cette PR tel quel : supprimer les 12 commits de A-312 (je ne sais pas pourquoi ils sont là mais il faut les enlever) et supprimer les anciennes icônes de Google+ (ainsi que le SCSS `.btn-google-plus`). Une fois que ceci est fait, je ne m'oppose pas à ce qu'on fusionne cette PR. :)

Sans que je ne m'y oppose, je ne comprends pas pourquoi tu souhaites faire une deuxième PR séparée pour les conventions graphiques de Google et Facebook. Cette PR n'est pas vraiment urgente en soit :)

helmasaur commented:
En fait, on en avait parlé à la fin fin juillet ou début août et j'avais émis la possibilité de faire cet ajout en attendant la modification pour que ça respecte les normes. Je voulais le faire dans une PR différente car celle-là date et aussi car on était presque à la publication d'une nouvelle version. Je voulais faire la modification pour Facebook et la mise aux normes de Google après.

Si j'en ai reparlé régulièrement, c'est parce que personne ne me répondais concernant ma demande dont on a discuté un peu plus tôt. Après, c'est plutôt normal de ne pas se souvenir de la discussion car c'était en plein durant les vacances.

Il n'y a donc aucun problème à ne pas avoir de réponse dans l'immédiat, mais il est vrai que j'en ai parlé régulièrement et on avait émis l'hypothèse de faire passer cette PR pour la version actuelle de ZdS début août en attendant la mise aux normes, car de toute façon, on ne respecte pas, pire encore, on utilise l'identité de Google+ qui est un échec pour eux.

Sinon, vu que le travail a été fait, autant faire les commits !

Je n'ai pas supprimé les logos de Google+ car je ne savais pas s'il fallait que je le fasse. Sinon, je ne sais pas comment supprimer des commits d'une PR.

AmauryCarrade commented:
> Si j'en ai reparlé régulièrement, c'est parce que personne ne me répondais concernant ma demande dont on a discuté un peu plus tôt. Après, c'est plutôt normal de ne pas se souvenir de la discussion car c'était en plein durant les vacances.

J'en suis désolé, ça n'a pas été très bien géré. Après, j'avoue que je ne m'en souvenais plus vraiment, de la demande initiale, donc les rappels sans le contexte me semblaient plutôt flous… d'où ma confusion sur ta demande.

> Il n'y a donc aucun problème à ne pas avoir de réponse dans l'immédiat, mais il est vrai que j'en ai parlé régulièrement et on avait émis l'hypothèse de faire passer cette PR pour la version actuelle de ZdS début août en attendant la mise aux normes, car de toute façon, on ne respecte pas, pire encore, on utilise l'identité de Google+ qui est un échec pour eux.

C'est vrai, c'est ce qui était prévu. Mais aujourd'hui, on a tout de même plus de temps — certes il est question de mettre cette PR dans la prochaine version, mais on n'est pas à une semaine près. Donc pourquoi ne pas tout faire, c'est vraiment ce que je me demande, vu le temps que ça prend. Ce pourquoi je me suis permis de te donner les images et les couleurs pour gagner du temps.

Sois-dit en passant, je pense que cette PR est une bonne occasion de cesser d'utiliser les _sprites_ et de passer sur des images SVG (ce que j'avais fait dans ma proposition). On aurait une meilleure qualité (c'est flagrant sur la capture plus haut), et c'est plus pratique.

> Sinon, vu que le travail a été fait, autant faire les commits !

Je peux, si tu me confirmes que tu le veux, mais je ne voulais pas te spoiler de quelque chose que tu désirais faire toi-même (d'où le partage des ressources uniquement). Surtout pour le temps que ça m'a demandé.

> Je n'ai pas supprimé les logos de Google+ car je ne savais pas s'il fallait que je le fasse.

On ne va plus s'en servir donc autant s'en séparer. Il faut juste vérifier que ce n'est pas utilisé ailleurs (je pense aux liens de partage), et à défaut, supprimer ces éventuels liens de partage vers Google+.

> Sinon, je ne sais pas comment supprimer des commits d'une PR.

Le plus simple c'est de faire un _rebase_ interactif. Depuis un état propre (pas de fichiers non-commités), fais `git rebase -i <hash d'un commit avant ceux d'A312>`. Ça va t'ouvrir un éditeur de texte avec tous les commits après celui dans la commande, chaque ligne ayant un format de ce genre : 

```
pick a77821d première ligne de la description du commit
```

Pour supprimer un commit, il suffit de _supprimer complètement la ligne du commit_ (ou des commits), puis d'enregistrer le fichier. Ensuite, une fois que tu auras vérifié que tout est bon dans le nouvel historique, il faut envoyer les modifications avec `git push --force` — précisément car un _rebase_ réécrit l'historique complètement.

Pour plus de détails, [la documentation de git est bien foutue (et en français)](https://git-scm.com/book/fr/v2/Utilitaires-Git-R%C3%A9%C3%A9crire-l%E2%80%99historique).

AmauryCarrade commented:
À la demande d'@Helmasaur, j'ai publié ce que j'avais tenté tantôt. J'ai aussi, dans ce commit, fait le nettoyage peu intéressant mais nécessaire des icônes : suppression des icônes `google-plus`, donc, mais aussi `google` et `foursquare` (??), inutilisées sur l'intégralité du site après vérification.

Je n'ai pas touché à l'icône `facebook` du sprite car elle est utilisée pour les boutons de partage.

AmauryCarrade commented:
À jour et nettoyé ; prêt pour QA.

Arnaud-D commented:
QA KO :x:  

---

J'ai remarqué un truc _sur mobile_ qui était déjà là plus ou moins avant, mais qui est plus visible désormais. C'est pas absolument bloquant, mais c'est moche sur mobile.

![image](https://user-images.githubusercontent.com/35631001/97040929-0129c580-156f-11eb-85d7-2d3768a8efe0.png)

Sur les deux boutons, on conserve des marges assez importantes, même avec des largeurs faibles. À la fin, le texte se met assez rapidement sur deux lignes, et ça dépasse du cadre...

AmauryCarrade committed:
Adaptation aux normes et correction mobile

- Usage des variables des normes graphiques.
- Correction de l'affichage sur mobile.

AmauryCarrade commented:
Le problème remonté a été corrigé, et la branche, mise à jour.

![image](https://user-images.githubusercontent.com/1417570/107234992-848afc00-6a24-11eb-9d89-02658e4433d1.png)

Situphen commented on assets/scss/base/_forms.scss:
```diff
@@ -256,37 +256,64 @@
+        span {
+            flex: 2;
+            text-align: center;
+            font-size:  $font-size-9;
```
```suggestion
            font-size: $font-size-9;
```

AmauryCarrade committed:
Perfectionnism is key

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

Situphen reviewed (approved):
**QA OK**
